addons/pkg-npm/common/src/main/java/org/commonjava/indy/pkg/npm/content/PackageMetadataGenerator.java

```diff
@@ -321,7 +321,7 @@ private boolean writePackageMetadata( List<StoreResource> firstLevelFiles, Artif
                 }
 
                 // Generate tarball url if missing
-                if ( versionMetadata.getDist() == null )
+                if ( versionMetadata.getDist() == null || !versionMetadata.getDist().getTarball().endsWith(packagePath.getTarPath()) )
```

Comment:
The original `dist` may contain other info that we should keep them as it is, ref: https://github.com/npm/registry/blob/main/docs/responses/package-metadata.md#dist

Or do we need to compare this, can we simply make it in `else` branch ? to avoid the potential null pointer exception.

Reply:
I have updated it to keep those extra data
And btw do we have linter or formatter?